test_conformance/api/test_kernels.cpp

```diff
@@ -704,3 +711,42 @@ REGISTER_TEST(negative_set_immutable_memory_to_writeable_kernel_arg)
 
     return TEST_PASS;
 }
+
+REGISTER_TEST(negative_invalid_arg_mem_obj)
+{
+    cl_int error = CL_SUCCESS;
+    clProgramWrapper program;
+    clKernelWrapper mem_obj_arg_kernel;
+
+    // Setup the test
+    error =
+        create_single_kernel_helper(context, &program, nullptr, 1,
+                                    &sample_mem_obj_size_test_kernel, nullptr);
+    test_error(error, "Unable to build test program");
+
+    mem_obj_arg_kernel = clCreateKernel(program, "mem_obj_size_test", &error);
+    test_error(error,
+               "Unable to get mem_obj_size_test kernel for built program");
+
+    std::vector<cl_uchar> mem_data(256, 0);
+    clMemWrapper buffer = clCreateBuffer(
+        context, CL_MEM_USE_HOST_PTR, mem_data.size(), mem_data.data(), &error);
+    test_error(error, "clCreateBuffer failed");
+
+    // Run the test - CL_INVALID_ARG_SIZE
+    error = clSetKernelArg(mem_obj_arg_kernel, 0, sizeof(buffer) * 2, &buffer);
```

Comment:
There's a bit of a gap in my C++ knowledge about the interaction of the `sizeof` opeator and user defined conversion operator, do you know or have you checked `sizeof(buffer)` calls `operator T() const { return object; }` then doing `sizeof()` on the returned object? It might be simpler to replace `sizeof(buffer)` with `sizeof(cl_mem)` to remove the distraction of this possibility.

Reply:
Good find, thanks - corrected!